bin/clover/src/cloud-control-funcs/code-gen/awsCloudControlCodeGenUpdate.ts

```diff
@@ -13,13 +13,40 @@ async function main(component: Input): Promise<Output> {
   const propUsageMap = JSON.parse(component.domain.extra.PropUsageMap);
   if (
     !Array.isArray(propUsageMap.createOnly) ||
-    !Array.isArray(propUsageMap.updatable)
+    !Array.isArray(propUsageMap.updatable) ||
+    !Array.isArray(propUsageMap.secrets)
   ) {
     throw Error("malformed propUsageMap on resource");
   }
 
   const payload = _.cloneDeep(component.domain);
 
+  // Push secrets to payload
+  for (
+    const {
+      secretKey,
+      propPath,
+    } of propUsageMap.secrets
+  ) {
+    const secret = requestStorage.getItem(secretKey);
+
+    if (!propPath?.length || propPath.length < 1) {
+      throw Error("malformed secret on propUsageMap: bad propPath");
+    }
+    if (secret) {
```

Comment:
Was noodling about how to do this with less integer indexes. If this appeals to you, feel free ... code is fine as is!

```suggestion
  // Grab the secret prop name and create its parent
  let secretName = propPath.pop();
  let secretParent = payload;
  for (const name of propPath) {
    secretParent[name] ??= {};
    secretParent = secretParent[name];
  }
  secretParent[secretName] = secret;
```

Reply:
Not sure how to expand a suggestion to include multiple lines :/